### PR TITLE
page layout

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR adjusts the page layout so that the sidebar (aside) is a sibling of the main element and is sticky (stays at the top of the viewport despite scrolling). Finishes work started in #299. 

If possible, it would probably be better to merge this into #299 and then review those together before merging it into main.

### :hammer_and_wrench: Detailed description

- removes wrapper inside body element
- adds grid css in `_layout.scss` for layout
- removed layout-related styles (as appropriate) from `app.scss`
- also removes icons from sidebar as added in an earlier PR, per designer feedback
- adds back in underlines to the sidebar, since we don't have icons there anymore
- adjusted some minor things with the header style so it looks nice.
- adjusted the sticky - it needed to be the `ul` inside of the `aside`, not the `aside` itself. 
- updated link names to use sentence case
- made the entire content width to be wider

There are other potential enhancements but I don't believe them to be essential at this time. Tracking them here: https://github.com/hashicorp/design-system/issues/313 

Screenshot:
CleanShot 2022-05-24 at 14 21 46

Tests pass: 
CleanShot 2022-05-24 at 14 20 43

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Comment by didoo** [comment]
@MelSumner can you sync both the branches (`melsumner/dummy-sidebar` and `melsumner/sidebar-page-layout`) with `main`? I see a lot of files that are not related to the PR and I suspect is because one of the two branches are out of sync

**Comment by MelSumner** [comment]
@didoo I have another PR after this one to dynamically make the list of links in the sidebar since we decided to take out the icons, but I am a little bit stuck on it so I might need to pair. If you're okay with it, I'd like to get the static thing merged and then PR the dynamic thing separately.

**Review by didoo — COMMENTED** [review]
@MelSumner I have added a few comments here and there, but probably is more efficient for both to have a pairing session and go through the code together. I'll book some time today.

File: packages/components/tests/dummy/app/styles/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			"footer footer footer footer";
+		grid-template-rows: auto auto 1fr auto;
+		grid-template-columns:
+			1fr
+			minmax(12em, calc(var(--content-width) * 0.2))
+			minmax(30em, calc(var(--content-width) * 0.8))
+			1fr;
+	}
+
+}
```
[the remark this answers is not in the document]
Comment (MelSumner) [reply packages/components/tests/dummy/app/styles/_layout.scss]: Tracking this here: https://github.com/hashicorp/design-system/issues/313

File: packages/components/tests/dummy/app/styles/_typography.scss [hunk 2]
```diff
@@ -27,6 +27,7 @@
     @include dummyFontSize(2.5rem);
     font-weight: 600;
     line-height: 1.2;
+    margin-top: 0.75rem; // makes it seem balanced visually with sidebar
```
[the remark this answers is not in the document]
Comment (MelSumner) [reply packages/components/tests/dummy/app/styles/_typography.scss:30]: tracking this here: https://github.com/hashicorp/design-system/issues/313

**Comment by Dhaulagiri** [comment]
Let's bring this into the base branch for this work

**Comment by Dhaulagiri** [comment]
Let's bring this into the base branch for this work

**Pull request merged**